modules-js/react-fleet/src/elements/inputs/FileInput.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import { VISUALLYHIDDEN } from '../../utilities/css';
+
+import { MEDIA_SMALL, OPTIMISTIC_BLUE } from '../../utilities/constants';
+
+import CloseButton from '../buttons/CloseButton';
+
+
+const INPUT_CONTAINER_STYLE = css(`
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+
+  ${MEDIA_SMALL} {
+    flex-wrap: nowrap;
+  }
+  
+  label {
+    margin-bottom: 0.5em;
+
+    ${MEDIA_SMALL} {
+      margin-bottom: 0;
+    }
+  }
+  
+  div {
+    flex-basis: 80%;
+        
+    ${MEDIA_SMALL} {
+      flex-basis: auto;
+      padding-left: 1em;
+    }
+  }
+`);
+
+const LABEL_FOCUSED_STYLE = css({
+  outline: `3px solid ${OPTIMISTIC_BLUE}`,
+  outlineOffset: '1px'
+});
+
+// todo: safari is not ever picking up this styling?
+const FILE_PREVIEW_STYLE = css({
+  position: 'relative',
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis'
+});
+
+const DELETE_BUTTON_STYLE = css({
+  marginLeft: '0.2em'
+});
+
+const DEFAULT_PREVIEW_TEXT = 'No file selected';
+
+
+interface Props {
+  name: string;
+  title: string;
+  fileTypes: string;
```

Comment:
Should this be an array because it’s “types”?

Reply:
I was torn on this; the native `<input type="file" />` expects a comma-separated string, but an array is a more logical structure. Since you also called it out, I'll modify.